### PR TITLE
Refactor inference manager code to save correct files

### LINKED ISSUE #862: [BUG] final_preds_and_avg_probs.csv file is not generated after inference
### To Reproduce
1. Use the data `BreastMNIST from https://medmnist.com/'`
2. Run command `'python /content/GaNDLF/gandlf_run -c /content/config.yaml -i /content/medmnist/dataset/test_path.csv -m /content/model/ -t False -d cuda'`

### Expected behavior
After running inference for the classification task following .csv files must be generated:
- output_predictions.csv
- final_preds_and_avg_probs.csv
- logits.csv

But final_preds_and_avg_probs.csv and logits.csv are not generating.

### Environment information
<!-- Put the output of the following command:
python ./gandlf_debugInfo

      GANDLF version: 0.0.20-dev
      Git hash: 50f369f9671f5b618e9e50568b349a419eca6a65
      Platform: Linux-6.1.58+-x86_64-with-glibc2.35
      Machine: x86_64
      Processor: x86_64
      Architecture: 64bit ELF
      Python environment:
        Version: 3.10.12
        Implementation: CPython
        Compiler: GCC 11.4.0
        Build: main Nov 20 2023 15:14:05

### PR DESCRIPTION
**Pull request by Geeks-Sid**
Fixes #862 

## Proposed Changes
- Ensured that n_folds is not used to trigger logit saving

## Checklist

- [x] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [x] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [x] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [x] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [x] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [x] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [x] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].

### CODE REVIEW

**Review by sarthakpati — APPROVED** [review]
Fix is confirming as expected. Merging.
